File: rename.py
#! /usr/bin/python
# coding:utf-8
import os
folder=raw_input('输入作业文件夹名:')
path='/Users/y4m1n/Downloads/'+folder
last_name=['方','孔','杨','胡','牛','王','陈','张','宋','董','付',
           '李','王','方','卫','孙','崔','文','钟','吴','唐','俞',
           '曾','刘','连','于','潘','黄','罗','苏','马','程','郑','蔡']
# 遍历目录内文件
for file in os.listdir(path):
    # 找有后缀的
    if file.find('.')>0:
        postfix = file.split('.')[-1]
        # 去空格，去掉后缀，去掉多余的“号”
        file_name = file.replace(" ","").split('.')[0].replace('号','')
        # 修改不规范连结符号
        if file_name.find('_') or file_name.find('+') or file_name.find('－'):
            file_name=file_name.replace('_','-')
            file_name=file_name.replace('+','-')
            file_name=file_name.replace('－','-')
        for name_part in file_name.split('-'):
            # 找学号
            if name_part.isdigit():
                id = name_part
                # 学号没写两位的则只保留两位
                if len(id) > 2:
                    id = id[-2:]
            # 找姓名
            elif name_part[:3] in last_name:
                name = name_part
        # 新名字
        new_name=str(id) + '-' + name + '.' + postfix
        file_path = os.path.join(path, file)
        new_file_path = os.path.join(path, new_name)
        # 改名
        os.rename(file_path, new_file_path)

File: rename.rb
#! /usr/local/bin/ruby
# Work with family_names.txt
family_names = File.read(File.join(File.dirname(__FILE__),'family_names.txt')).split(' ')
teacher_names = ['老师', '大嘴']
# 可根据人数调整此参数
num_of_classmates = 47
id_list = (1..num_of_classmates).to_a
# 退出的同学学号
dropouts = [29, 45, 34]
id_list = id_list - dropouts
Dir.glob("#{ARGV[0]}/**/*") .each do |file_name|
  if file_name.scan(/\d{1,}号/)[0].length==2:
    id = '0'+file_name.scan(/\d{1,}号/)[0][0]
  else
    id = /\d{2,}/.match(file_name).to_s[-2..-1]
  puts 'WARNING: repeated id: ',file_name unless id_list.include?(id.to_i)
  id_list.delete(id.to_i)
  name = file_name.delete('号').scan(/[\u4e00-\u9fa5]{2,}/).select { |text| (text.length < 4 ) && (family_names.include?(text[0])) unless teacher_names.include?(text[1..-1]) }
  puts 'WARNING: more than one names got: ', name if name.size > 1
  new_name = file_name.sub(File.basename(file_name), id.to_s + '-' + name.join('-') + File.extname(file_name))
  File.rename(file_name, new_name)
end
if id_list.empty?
  puts '作业已收齐'
else
  puts '没交作业的同学学号：'
  puts id_list.to_s
end
